x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -245,21 +241,16 @@ func (c *runCmd) runStepFunc(
 			return err
 		}
 		// TODO: implement readonly for now just don't charge for gas
-		response, _, err := programExecuteFunc(ctx, c.log, db, programAddress, params[1:], method, math.MaxUint64)
-		if err != nil {
+		output, _ := programExecuteFunc(ctx, c.log, db, programAddress, params[1:], method, math.MaxUint64)
```

Comment:
It's here that you should deal with an error return by `programExecuteFunc`.

Reply:
Done by the add of the `resultToOutput` function which gets passed the output as well as the error.